_studio/shared/umc/codec/mpeg2_dec/src/umc_mpeg2_utils.cpp

```diff
@@ -0,0 +1,488 @@
+                in->mfx.CodecProfile == MFX_PROFILE_MPEG2_MAIN ||
+                in->mfx.CodecProfile == MFX_PROFILE_MPEG2_HIGH ||
+                in->mfx.CodecProfile == MFX_PROFILE_MPEG1 ||
+                in->mfx.CodecProfile == 0))
+            {
+                out->mfx.CodecProfile = 0;
+                return MFX_ERR_UNSUPPORTED;
+            }
+
+            if ((in->IOPattern & MFX_IOPATTERN_OUT_SYSTEM_MEMORY) || (in->IOPattern & MFX_IOPATTERN_OUT_VIDEO_MEMORY))
+                out->IOPattern = in->IOPattern;
+            else if (MFX_PLATFORM_SOFTWARE == core->GetPlatformType())
+                out->IOPattern = MFX_IOPATTERN_OUT_SYSTEM_MEMORY;
+            else
+                out->IOPattern = MFX_IOPATTERN_OUT_VIDEO_MEMORY;
+
+            if (!IsHWSupported(core, in))
+            {
+                return MFX_ERR_UNSUPPORTED;
+            }
+        }
+
+        return res;
+    }
+
+    // Table 6-4 – frame_rate_value
+    void GetMfxFrameRate(uint8_t frame_rate_value, mfxU32 & frameRateN, mfxU32 & frameRateD)
+    {
+        switch (frame_rate_value)
+        {
+            case 0:  frameRateN = 30;    frameRateD = 1;    break;
+            case 1:  frameRateN = 24000; frameRateD = 1001; break;
+            case 2:  frameRateN = 24;    frameRateD = 1;    break;
+            case 3:  frameRateN = 25;    frameRateD = 1;    break;
+            case 4:  frameRateN = 30000; frameRateD = 1001; break;
+            case 5:  frameRateN = 30;    frameRateD = 1;    break;
+            case 6:  frameRateN = 50;    frameRateD = 1;    break;
+            case 7:  frameRateN = 60000; frameRateD = 1001; break;
+            case 8:  frameRateN = 60;    frameRateD = 1;    break;
+            default: frameRateN = 30;    frameRateD = 1;
+        }
+        return;
+    }
+
+    // Table 8-2 – Profile identification
+    mfxU8 GetMfxCodecProfile(uint8_t profile)
+    {
+        switch (profile)
+        {
+            case 5:
+                return MFX_PROFILE_MPEG2_SIMPLE;
+            case 4:
+                return MFX_PROFILE_MPEG2_MAIN;
+            case 1:
+                return MFX_PROFILE_MPEG2_HIGH;
+            default:
+                return MFX_PROFILE_UNKNOWN;
+        }
+    }
+
+    // Table 8-3 – Level identification
+    mfxU8 GetMfxCodecLevel(uint8_t level)
+    {
+        switch (level)
+        {
+            case 10:
+                return MFX_LEVEL_MPEG2_LOW;
+            case 8:
+                return MFX_LEVEL_MPEG2_MAIN;
+            case 6:
+                return MFX_LEVEL_MPEG2_HIGH1440;
+            case 4:
+                return MFX_LEVEL_MPEG2_HIGH;
+            default:
+                return MFX_LEVEL_UNKNOWN;
+        }
+    }
+
+    bool DARtoPAR(uint32_t width, uint32_t height, uint32_t dar_h, uint32_t dar_v,
+                uint16_t & par_h, uint16_t &par_v)
+    {
+        uint32_t simple_tab[] = {2,3,5,7,11,13,17,19,23,29,31,37,41,43,47,53,59};
```

Comment:
Euclidean algorithm for ratio simplification can be used.

Reply:
This is copy paste from legacy implementation,

Reply:
ok, improve in future